docs/source/start/declarative.rst

```diff
@@ -68,9 +68,41 @@ containing YAML, wrap it in :external:class:`io.StringIO`:
 Format description
 ==================
 
-The expected YAML follows a simple format, where a parent object (i.e. an
-object that already exists in the model) is selected, and one or more of three
-different operations is applied to it:
+Metadata
+--------
+.. versionadded:: 0.5.16
+   Added metadata document to the declarative modelling YAML.
+
+The metadata section is optional and has the following format:
+
+.. code-block:: yaml
+
+   model:
+     version: 2fd4e1c67a2d28fced849ee1bb76e7391b93eb12
+     url: https://example.com/model.git
+     entrypoint: path/to/model.aird
+   written_by:
+     capellambse_version: 1.0.0
+     generator: Declarative Modelling Generator 1.0.0
+
+It gives information about which model the declarative modelling YAML file
+wants to change, with which capellambse version and generator it was written
+with. A versioned model (e.g. using a git URL) can be uniquely identified by
+its revision and the repository ``url``. These values are checked against the
+``model.info`` during ``decl.apply`` when the ``strict`` parameter is ``True``
+(``False`` per default).
+
+Instructions in the declarative modelling YAML may need additional instructions
+for references such that the changes are correctly displayed in Capella. E.g.
```

Comment:
> such that the changes are correctly displayed in Capella

This is not the reason; the reason is that the model becomes broken and useless (both in Capella and CapellaMBSE) if this isn't done.

However, as CapellaMBSE will always do these checks implicitly anyways and remove such dangling references automatically (as soon as the relevant PR is merged), this entire sentence doesn't really make a whole lot of sense anymore. The "additional instructions" mentioned here are only needed if the YAML metadata specified the "explicit" referencing style (which isn't implemented yet, and probably won't be implemented for a while).